File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }

        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        int maxDepth = 0;

        while (!queue.isEmpty()) {
            int levelSize = queue.size(); // Store the current size of the queue

            // Process all nodes at the current level
            for (int i = 0; i < levelSize; i++) {
                TreeNode node = queue.poll();

                if (node.left != null) {
                    queue.add(node.left);
                }

                if (node.right != null) {
                    queue.add(node.right);
                }
            }

            maxDepth++; // Increment depth after processing all nodes at the current level
        }

        return maxDepth;
    }
}

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root: return 0

        queue = deque()

        queue.append(root)

        max_depth = 0

        while queue:
            for i in range(len(queue)):
                current_level = queue.popleft()

                if current_level.left: 
                    queue.append(current_level.left)

                if current_level.right:
                    queue.append(current_level.right)
            max_depth+=1
        

        return max_depth
        